File: gae/app/models/Product.java
package models;

import play.modules.siena.EnhancedModel;
import siena.Generator;
import siena.Id;
import siena.Max;
import siena.Model;
import siena.Query;
import siena.Table;

import com.google.gson.Gson;


@Table("products")
public class Product extends EnhancedModel {

    @Id(Generator.AUTO_INCREMENT)
    public Long id;
    
    public String code;

    public String name;

    public String thumb;

    @Max(1000)
    public String description;

    public double price;

    public double special;

    public int rating;

    public int reviews;
    
    public String toString() {
      return name;
    }
    /*
    public static Product findById(String name) {
      return Model.getByKey(Product.class, name);
    }

    public static Query<Product> all() {
      return Model.all(Product.class);
    }
    */

    public static Product findByCode(String code) {
      return Product.all().filter("code", code).get();
    }
}

File: main/app/models/Product.scala
package models

import play.api.db._
import play.api.Play.current

import anorm._
import anorm.SqlParser._

case class Product(product_id: Pk[Long], name: String, thumb: String, description: String, price: Double, special: Double = 0)

object Product {
  // -- Parsers
  
  /**
   * Parse a Product from a ResultSet
   */
  val simple = {
    get[Pk[Long]]("product.id") ~
    get[String]("product.name") ~
    get[String]("product.thumb") ~
    get[String]("product.description") ~
    get[Double]("product.price") map {
      case id~name~thumb~description~price => Product(id, name, thumb, description, price)
    }
  }
  
  // -- Queries
    
  /**
   * Retrieve a Product from id.
   */
  def findById(id: Long): Option[Product] = {
    DB.withConnection { implicit connection =>
      SQL("select * from product where id = {id}").on(
        'id -> id
      ).as(simple *).headOption
    }
  }

  /**
   * Retrieve Products by id's.
   */
  def findByIds(ids: Seq[Long]): List[Product] = {
    DB.withConnection { implicit connection =>
      val idString = ids.map(_.toString)
      SQL("select * from product where id in (" + ids.map("{a" + _ + "}").reduceLeftOption(_ + ", " + _).
          getOrElse("") +")").onParams(ids.map(anorm.toParameterValue(_)) : _*).as(simple *)
    }
  }

  /**
   * Retrieve all Products.
   */
  def findAll = {
    DB.withConnection { implicit connection =>
      SQL("select * from product").as(simple *)
    }
  }

  /**
   * Create Product.
   */
  def create(product: Product): Product = {
    DB.withConnection { implicit connection =>
      SQL(
        """
insert into product(name, thumb, description, price) values (
{name}, {thumb}, {description}, {price}
)
"""
      ).on(
//        'id -> product.product_id,
        'name -> product.name,
        'thumb -> product.thumb,
        'description -> product.description,
        'price -> product.price
      ).executeUpdate()
      
      product
      
    }
  }
}

